crates/types/src/block_impl.rs

```diff
@@ -126,3 +130,20 @@ impl BlockPayload for VIDBlockPayload {
             .collect()
     }
 }
+
+/// A [`BlockHeader`] that commits to [`VIDBlockPayload`].
+#[derive(PartialEq, Eq, Hash, Clone, Debug, Deserialize, Serialize)]
+pub struct VIDBlockHeader {
+    /// Block number.
```

Comment:
Do we also need to allow the application to define fields if they want to?  @jbearer ?

Reply:
Yes. I think the header should be an app-specific type, with a trait for accessing the fields needed by consensus (e.g. payload commitment)

Reply:
I think so! `VIDBlockHeader` is one implementation of `BlockHeader` but there could be other implementations with different fields.